src/dataset-builder/DomainCriteriaSelector.ts

```diff
@@ -59,19 +64,41 @@ export const saveSelected =
 
 export const DomainCriteriaSelector = (props: DomainCriteriaSelectorProps) => {
   const [rootConcepts, loadRootConcepts] = useLoadedData<GetConceptsResponse>();
+  const [hierarchyConcepts, loadHierarchyConcepts] = useLoadedData<GetConceptsHierarchyMapResponse>();
   const { state, onStateChange, datasetId, getNextCriteriaIndex } = props;
+  const [hierarchy, setHierarchy] = useState(new Map<number, Concept[]>());
+  const [isHierarchyLoaded, setIsHierarchyLoaded] = useState(false);
   useOnMount(() => {
-    void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));
+    const selectedConcept = state.selectedConcept;
+    if (selectedConcept) {
+      void loadHierarchyConcepts(() => DataRepo().dataset(datasetId).getConceptsHierarchy(selectedConcept));
+    } else {
+      // get me the children of this concept id
+      void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));
+    }
   });
-  return rootConcepts.status === 'Ready'
+
+  useEffect(() => {
+    if (rootConcepts.status === 'Ready') {
```

Comment:
so this (case1 ) correlates with: ln 77 void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));

and (case2) else if (hierarchyConcepts.status === 'Ready') correlates with ln 74  void loadHierarchyConcepts(() => DataRepo().dataset(datasetId).getConceptsHierarchy(selectedConcept));

Is that right? Why do we still need to set the hierarchy for case 1? It isn't going to be rendered, right? It's only if a concept is selected/opened that the hierarchy view is displayed. I just want to make sure I understand what is going on here!

Reply:
Yes! We are using the statuses of the AJAX hooks. If rootConcepts.status === 'Ready' this means that `openedConcept` hasn't been passed to DomainCriteriaSelector and the getConcepts() ajax hook has been called. In the other case, if hierarchyConcepts.status === 'Ready' this means `openedConcept` has been passed to DomainCriteriaSelector and we are going to get the hierarchy of this selected concept.